aten/src/ATen/native/Math.h

```diff
@@ -1512,6 +1513,11 @@ calc_i1(T _x) {
   return (_x < T{0.0}) ? -out : out;
 }
 
+// Upcast bfloat16/half input to float for numerical accuracy purposes
+inline c10::BFloat16 calc_i1(c10::BFloat16 a) { return calc_i1(static_cast<float>(a)); }
```

Comment:
Why aren't these just template instatiations?

Reply:
Because https://github.com/pytorch/pytorch/blob/929797dedbf23376123ce95230c01a7e3b71e130/aten/src/ATen/native/Math.h#L1466 and neither Half nor BF16 are considering floating point types. Which reminds me of another simple BE task (enable_if_t :P )